File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/ctrlauth/Simple/VB/SimpleInnerContent.vb
Option Strict Off

Imports System
Imports System.Web
Imports System.Web.UI

Namespace SimpleControlSamples

    Public Class SimpleInnerContentVB : Inherits Control

       Protected Overrides Sub Render(Output As HtmlTextWriter)

           If HasControls() And TypeOf Controls(0) Is LiteralControl
              Dim Ctrl As LiteralControl = Controls(0)
              Output.Write("<H2>Your Message: " & Ctrl.Text & "</H2>")
           End If

       End Sub

    End Class

End Namespace

File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/ctrlauth/Simple/SimpleInnerContent.cs
using System;
using System.Web;
using System.Web.UI;

namespace SimpleControlSamples {

    public class SimpleInnerContent : Control {

       protected override void Render(HtmlTextWriter output) {

           if ( (HasControls()) && (Controls[0] is LiteralControl) ) {
              output.Write("<H2>Your Message: " + ((LiteralControl) Controls[0]).Text + "</H2>");
           }
       }
    }    
}